Modules/Authentication/RequireRoleAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿using GenHTTP.Modules.Authentication.Roles;
+using GenHTTP.Modules.Reflection;
+
+namespace GenHTTP.Modules.Authentication;
+
+/// <summary>
+/// When annotated on a service method, requests will only be allowed
+/// if the authenticated user has the specified roles.
+/// </summary>
+/// <param name="roles"></param>
+[AttributeUsage(AttributeTargets.Method)]
+public class RequireRoleAttribute(params string[] roles) : InterceptWithAttribute<RoleInterceptor>

```

Comment:
I'm still thinking whether single-use attribute class with collection of roles is more clear than multi-use attribute class with single designated role.

I.e.:
```C#
[RequireRole("Editor", "Viewer")]
public async ValueTask DoSomething() ...
```

vs

```C#
[RequireRole("Editor")]
[RequireRole("Viewer")]
public async ValueTask DoSomething() ...
```

IMO the later is more maintainable and clear (my preference is more simple lines over one long line). Also as-is now, the singular attribute name suggests it expects single role.
That said I don't have a strong preference.

Reply:
I typically would prefer the first variant but that also depends on your role model I guess. Your example mixed different kind of roles (user tiers, packages, etc.), so I probably want them on a seperate line. Nevertheless, with the current approach you can do both (probably need to add another test for the second variant), so I will leave it like this I guess.

Reply:
FYI For the second variant you need to specify AttributeUsage.AllowMultiple.

Reply:
Of course ... thanks!